### ISSUE TITLE
Drop usage of pkg_resources

### ISSUE THREAD

**Issue by kasium** [marshmallow-code/apispec-webframeworks#108]
`src/apispec_webframeworks/__init__.py` uses pkg_resources which is deprecated. What about

- Drop python 3.6 and 3.7 support
- Replace pkg_resources with importlib.metadata

I'm happy to provide the necessary PR

**Comment by lafrech** [comment]
Hi. This would be awesome!

I don't have a strict objection to pyproject.toml but I'd rather keep things here consistent with other marshmallow-code packages.

We should fix CI before, or in this PR, by moving to Github Actions (see https://github.com/marshmallow-code/marshmallow/pull/2098). @sloria do you think you could add `PYPI_API_KEY` (and/or make me maintainer) so that we can proceed?

Thanks!

**Comment by kasium** [comment]
@lafrech I'm happy to try the GitHub Action migration first. I guess somewhen next week I'll find some time to try that

**Comment by lafrech** [comment]
Le Sun, 06 Aug 2023 04:11:19 -0700,
kasium ***@***.***> a écrit :

> @lafrech I'm happy to try the GitHub Action migration first. I guess
> somewhen next week I'll find some time to try that

Great. We only need the PYPI API key to release but we can start
working on the transition to GHA without it. It should be relatively
easy to copy the configuration from the other marshmallow repos.

-- 
Jérôme

**Comment by lafrech** [comment]
@sloria I'm talking about PYPI ownership. I don't see those in my list: https://pypi.org/user/lafrech/. So I can't create the API key.

**Comment by sloria** [comment]
@lafrech oh ok, i've given you access to apispec-webframeworks. 

i can't give access to marshmallow-oneofschema since i'm not owner. @maximkulkin could you please make me and @lafrech owners of the pypi package here: https://pypi.org/manage/project/marshmallow-oneofschema/collaboration/

**Comment by lafrech** [comment]
Thanks. I just added `PYPI_API_TOKEN` to this repo.

**Comment by lafrech** [comment]
- [x] Drop python 3.6 and 3.7 support
- [ ] Replace pkg_resources with importlib.metadata
- [x] Update versions used by pre-commit
- [ ] Use modern packging based on pyproject.toml => Not sure. Unless we want to do this for all marshmallow repos. Consistency first.
- [ ] Add support for python3.10/11/12 => I added 10 and 11. Let's add 3.12 as in https://github.com/marshmallow-code/marshmallow/pull/2188. Also, I added this repo to pre-commit CI so after we check it actually works we can modify workflow to only run lint on tags as in marshmallow. 

Thank you for you help on this @kasium.

**Comment by lafrech** [comment]
Other TODOs:

- [ ] setup.cfg: remove "universal" line
- [ ] setup.cfg: review flake8 config, remove "select" and question other lines (compare with marshmallow and other libs here)

- [ ] improve the way minimal versions of dependencies are provided (see https://github.com/marshmallow-code/webargs/blob/dev/setup.py for an example) and test minimum dependencies in a dedicated test to catch compatibility issues (https://github.com/marshmallow-code/webargs/blob/dev/tox.ini)

**Comment by Quentin-Pinot** [comment]
Hello, 

Regarding the TODO:  "Replace pkg_resources with importlib.metadata" i just made one PR about it.

**Comment by sloria** [comment]
@kasium good catch. updated in https://github.com/marshmallow-code/apispec-webframeworks/commit/980ac9d56e50efafb4d191bec51067f844c3cb7e
